cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/worker/TaskWorkerServiceLauncher.java

```diff
@@ -152,7 +161,9 @@ public void run() {
             twillPreparer = ((StatefulTwillPreparer) twillPreparer)
               .withStatefulRunnable(TaskWorkerTwillRunnable.class.getSimpleName(), false,
                                     new StatefulDisk("task-worker-data", diskSize,
-                                                     cConf.get(Constants.CFG_LOCAL_DATA_DIR)));
+                                                     cConf.get(Constants.CFG_LOCAL_DATA_DIR)))
+              .withDependentRunnables(TaskWorkerTwillRunnable.class.getSimpleName(),
```

Comment:
Why this method is inside the `StatefulTwillPreparer`? Seems like it is applicable to non stateful case as well.

Reply:
It is applicable to non stateful cases as well. I initially put it in the StatefulTwillPreparer block for simplicity.
Fixed now.